### ISSUE TITLE
MonoDevelop overwrites project GUID to be lowercase instead of uppercase

### ISSUE THREAD

**Issue by NatElkins** [mono/monodevelop#5976]
**Steps to repro:**

1. Check out https://github.com/microsoft/visualfsharp
2. Open FSharp.sln using MonoDevelop.
3. Rename any project.
4. In repo, do `git status`
5. Observe that there are a bunch of updates similar to the following:
```
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "FSharp.Tests.FSharpSuite", "tests\fsharp\FSharp.Tests.FSharpSuite.fsproj", "{C163E892-5BF7-4B59-AA99-B0E8079C67C4}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "FSharp.Tests.FSharpSuite", "tests\fsharp\FSharp.Tests.FSharpSuite.fsproj", "{C163E892-5BF7-4B59-AA99-B0E8079C67C4}"
```
Note that `F2A71F9B-5D33-465A-A702-920D77279786` gets changed to `f2a71f9b-5d33-465a-a702-920d77279786`.

I believe that MonoDevelop should not change the casing of these GUIDs.

> VS bug [#686313](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/686313)

**Comment by knocte** [comment]
what version of MonoDevelop are you using?

**Comment by NatElkins** [comment]
7.5 (build 1255)

I see it replaces other project GUIDs as well, for example:
```
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ProjectSystem.Base", "vsintegration\src\FSharp.ProjectSystem.Base\Project\ProjectSystem.Base.csproj", "{B700E38B-F8C0-4E49-B5EC-DB7B7AC0C4E7}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ProjectSystem.Base", "vsintegration\src\FSharp.ProjectSystem.Base\Project\ProjectSystem.Base.csproj", "{B700E38B-F8C0-4E49-B5EC-DB7B7AC0C4E7}"
```

**Comment by mrward** [comment]
The problem with the .NET Core project type guid 9A19103F-16F7-4668-BE54-9A1E7A4F7556 being changed back to the C# project type guid has been fixed in master and 7.7 - https://github.com/mono/monodevelop/pull/5864

I think the problem with the F# project type guid is that it is defined in lowercase in the F# addin - https://github.com/mono/monodevelop/blob/926494d80a67df78c3abb0b147616a26de8135d3/main/external/fsharpbinding/MonoDevelop.FSharpBinding/FSharpBinding.addin.xml#L40 - it should be using uppercase to be consistent with Visual Studio on Windows.

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.626 (master)

Pull Request #6109 merged by: Jason Imison
Author: knocte
Commit: e50935be089528b773d112806036a0601736f4c2 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.197 (release-8.0)

Pull Request #6109 merged by: Jason Imison
Author: knocte
Commit: e50935be089528b773d112806036a0601736f4c2 (mono/monodevelop)
